### DISCUSSION TITLE
hackintosh/updates/2022/01/09/alder-lake

### DISCUSSION THREAD

**Discussion by giscus** [dortania/dortania.github.io#29] [Comments]
# hackintosh/updates/2022/01/09/alder-lake

Last year, when I presented the Z590 nuances here on Dortania (main paper, amendments, sample configuration), I was rather worried that with the introduction of Intel Z690 architecture we will be out of luck. Today I am confident that it is not the case. While it is far from perfect and while it is harder to make Alder Lake run on the kernel side, after several days of experiments with @vandroiy2013 we were able to make it. Here comes what one needs to be aware of when building a Z690-based setup. As usual, the previous knowledge of Z490 and Z590 still applies, and I strongly advise you to acquaint yourself with these platforms first. Please keep in mind that up to this moment we have only tested one ASUS motherboard and although the described issues should normally apply to the other vendors, we can neither guarantee that we spotted them all, nor say that we performed all the tests even with the board we had at hand. BIOS Given that we waited a bit longer for a news post this time, Alder Lake firmware were updated to be more acceptable. Apple RTC issues are finally resolved at least in ASUS boards. While one still obviously needs an AWAC SSDT table, DisableRtcChecksum or RTCMemoryFixup.kext are no longer required. CFG Lock may not be configurable in preferences on ASUS boards. This is an obvious BIOS bug, although it may not cause boot failures. We had to unlock it manually through the Shell method described in OpenCore Reference Manual. Unlocking with ControlMsrE2 was unsuccessful, but we did not investigate it much. XMP works at least with DDR5 we had at hand, but there were reports of no issues with DDR4 as well. While macOS does not name DDR5 as DDR5 in the profiler, this nuance is purely cosmetic. We did not have PCIe 5.0 hardware to ensure optimal performance, but there were no issues with PCIe 4.0 and 3.0 including Resize Bar support handled by the ResizeAppleGpuBars quirk. We did not have native Thunderbolt support in our board, but Z690 Thunderbolt firmware is also part of the main firmware, and thus can hardly be modified. Flashed PCIe cards should work fine though. ASUS boards have a new aggregate GOP instance, which causes black screen during macOS first stage. This is addressed in the ProvideConsoleGop quirk starting with OpenCore 0.7.6. Most firmware dropped Processor-based CPU definition in ACPI and switched to Device-based definition, which is not recognised by macOS. To workaround this one needs to use the SSDT-PLUG-ALT ACPI table. CPU tuning Unlike the previous generation, only Alder Lake CPUs are supported in Z690 boards. Furthermore, the new CPUs utilise a heterogenous architecture, similarly to big.LITTLE found in ARM. There is a trade-off between using different core kinds and getting better multicore performance and sticking with performance cores. In case you enable the efficiency cores be aware: All cores are to support the same instruction set on x86, which means AVX-512 is disabled. Ring bus frequency is lowered to match the efficiency cores (3.6 vs 5.0), which slowdowns the overall performance. XNU scheduler does not distinguish core types in macOS and will work no better than current Windows 10 in terms of core assignment. More patches are required for XNU when using the efficiency cores, though handled automatically by the ProvideCurrentCpuInfo quirk starting with OpenCore 0.7.7. Enabling the efficiency cores will require RestrictEvents CPU name spoofing functionality due to large core count. As for the issues not related to the hybrid architecture: Alder Lake CPUs also require CPUID virtualisation to boot. Partial XCPM compatibility is available, but frequency vector tuning will be required. Thermal management works. While Intel Power Gadget is rather buggy and not recommended in production, it mostly works, so do SuperIO and SMCProcessor. Audio and graphics Just like Z590, Z690 boards support USB and analog audio. Analog audio patches were contributed by @R-a-s-c-a-l and are present starting with AppleALC 1.6.7. USB audio is supported by AppleALCU out of the box. Please note that USB audio may need separate firmware updates provided by the motherboard vendor (like on ASUS Z590 series, which initially had sporadic crackling noise). Builtin IGPU is not supported by macOS starting with Rocket Lake. In case it is needed for other operating systems, it is recommended to disable it with class-code = <FF FF FF FF>. This applies to all uses of disable-gpu, which has been proven to be unreliable. One can also disable it in firmware preferences. Discrete GPUs are supported without any limitations including DRM and digital audio. USB No change from Z590: Needs usual care with port mapping to get working sleep support. USBWakeFixup is needed to fix keyboard wakeup support, but may cause compatibility issues with Bluetooth. Overclocking RAM is known to cause USB issues, but we were unable to spot anything negative with XMP so far. Networking and wireless Intel 2.5 Gbps adapters work out of the box through the DEXT drivers at least on ASUS boards. Intel Wi-Fi works with AirportItlwm, but we had wake-up issues at the time we tested it. The issue is tracked here. Intel Bluetooth works with IntelBluetoothFirmware in macOS 11, but may not be functional in macOS 12. The issue is tracked here. As usual we recommend replacing Intel wireless modules with Apple ones whenever possible. Miscellaneous No change from Z590: S3 support works fine. FileVault 2 authenticated restart works fine. To summarise, Intel Z690 looks like a solid upgrade which brings better performance per watt. We can still run macOS on it with acceptable efficacy. — Vit

https://dortania.github.io/hackintosh/updates/2022/01/09/alder-lake.html

**Comment by yaduramauk** [comment]
Thank you for the info

**Reply by perez987** [reply]
Thank you very much!!!

**Reply by f0rdream** [reply]
Great job!

**Reply by claudiusraphael** [reply]
I am researching hardware for quite some time for an actual yet low-end setup, to
- get a baseline of performance on an actual architecture for software development
- see how low the budget can go, while still guaranteeing upgradability
- be as cross-platform as possible.

The Z690/S1700 fullfills the requirements so far, as it supports DDR4/5, Thunderbolt, PCIe5.0, 128GB RAM. Which is by far the widest spectrum on the Intel-based platforms for Desktop/Mobile allowing Office/Gaming/Workstation setups.

After 2 decades of pure hatred for the Apple-Intel Combo i decided to give the M1 a chance, so my requirement is not Apple-specific, but i wish to have the possibility of comparing and optimizing development for Intel-based macs alongside the M1.

So last step in my research for components would be to checkout shortcomings.

I'm glad you succeeded in working it out in general!

I wonder if you author(s)/commenters can leave some notes specifically on:

+ Motherboard/Firmware regarding OpenCore/macos on AsRock vs. ASUS, or if there is an alternstive manufacturer i should have an eye on; I tend to be skeptical towards MSI and GigaByte, due to bad(dest) experiences especially for non-Windows setups and also hardware-stability, but i will consider them if there are reasonable suggestions.

CPU, namely:
- Pentium Gold G7400 ( https://www.intel.com/content/www/us/en/products/sku/219435/intel-pentium-gold-g7400-processor-6m-cache-3-70-ghz/specifications.html?wapkw=pentium%20g7400 )
- Celeron G6900 ( https://ark.intel.com/content/www/us/en/ark/products/96163/intel-celeron-processor-g6900-4m-cache-3-40-ghz.html )

These two build the lowest barrier according to my requirements, due to their iGPU (UHD Graphics 710), the cost (cheap for Intel products) and their virtualization capabilities and AVX, which are needed for hardware-passthrough via KVM/qemu, Windows-Subsystem for Android and WSL2/gWSL, Windows Containers and Sandbox.

Regarding macos: if i understood correctly, the iGPU will in no case be usable, which is fine, in fact useful in the case of running macos virtualized with gpu-passthrough.

Specifically for compatibility to older os versions  i bought:
- Nvidia Quadro K2000D, due to its compatibility with macos/windows/illumos
- Nvidia Quadro M2000, due to its compatibility with macos/windows/illumos

I consider getting a AMD Firepro W5100, which would be the stable upper limit for Windows XP specific usability, but i am not sure if thst would be usable with macos of any version.

The main question remaining regarding actual and futureproof GPU what would be a low-end baseline model for a discrete graphics card for so far Big Sur/Monterey in this setup? (preferably old/refurbished/used workstation graphics).

What i want to achieve is to support:
- actual/future low-end
- up to HighSierra/Mojave/Catalina for mid-/low-end
- and if possible get as low as it gets (Tiger?)

... to learn how macos developed over the years, user habits, performance of hardware and software, but also to be able to find a conclusion and how low i could go regarding compatibility with these old systems when developing new software.

RAM - In relation to that, especially for performance testing i wish to have the possibility of just using one DIMM to see how much the impact is versus Dual/Triple/... channel, so the question would be what is the lowest limit people actually use with macos and what is the general applied in actual size?

For the Workstation-use under Linux/BSD/Illumos/Windows-based Workloads i will use 32GB modules, or 16GB modules depending on the motherboard it all ends up on. 

For testing older macos/osx-releases directly i would like to have seperate modules at hand with the mentioned common sizing. What would you recommend?

STR - Regarding storage i have a wide range of drives at hand in the most used formats, from HDD (2GB to 8TB, some trays full) with RPM from 1600 to 15k, CF-II, eMMC, SSHD from 512GB to 2TB, SSD from 15 year old 2GB over all standard sizes to Samsung 2TB QVO 870 as the upper end. Regarding actual macos i guess i need to plan in a nvme drive in the near future?

Any recommendations that accomodate my requirements are more than welcome!

Wow, that's quite a wall of text.

Please feel free to move it elsewhere if necessary, but due to the specific constraint to the Z690 i chose to put it here.

Thanks for your work and for freely sharing your results!

**Comment by Octavian1973** [comment]
Very good work! Excellent achievements! Congratulations!
But when will you update, correctly and up to date, dortania github io OpenCore-Install-Guide site?

**Comment by ccnmxns** [comment]
FYI B660 is capable as well ;)

**Comment by developersiapp** [comment]
Is Alter Lake Intel Graphics 710,720,730  (iGPU) support to Big Sur/Monterey.

**Reply by ccnmxns** [reply]
Doubt that...

**Comment by NazarKalytiuk** [comment]
Do you plan to add Alder Lake to the installation guide?
